tests/admin_views/tests.py

```diff
@@ -5801,6 +5801,63 @@ def setUp(self):
+        )
+        self.selenium.get(self.live_server_url + reverse("admin:auth_group_add"))
+
+        box_ids = {
+            "from": '<div id="id_permissions_from_title" class="selector-available-title"><label for="id_permissions_from">Available permissions </label><p class="helptext">This is the list of available permissions. You may choose some by selecting them in the box below and then clicking the "Choose" arrow between the two boxes.</p></div>',
+            "to": '<div class="selector-chosen-title" id="id_permissions_to_title"><label for="id_permissions_to">Chosen permissions </label><p class="helptext">This is the list of chosen permissions. You may remove some by selecting them in the box below and then clicking the "Remove" arrow between the two boxes.</p></div>'
+        }
+
+        for box_id in box_ids:
+            box = self.selenium.find_element(By.ID, f"id_permissions_{box_id}")
+            self.assertEqual(box.get_attribute("aria-labelledby"), f"id_permissions_{box_id}_title")
+
+            box_helptext = self.selenium.find_element(By.ID, f"id_permissions_{box_id}_title")
+            self.assertEqual(box_helptext.get_attribute("outerHTML"), box_ids[box_id])
+
+    def test_FilteredSelectMultiple_filter_labels(self):
+        from selenium.webdriver.common.by import By
+
+        self.admin_login(
+            username="super", password="secret", login_url=reverse("admin:index")
+        )
+        self.selenium.get(self.live_server_url + reverse("admin:auth_group_add"))
+
+        filter_ids = {
+            "filter": '<label for="id_permissions_input"><span class="help-tooltip search-label-icon" aria-label="Type into this box to filter down the list of available permissions."></span></label>',
+            "filter_selected": '<label for="id_permissions_selected_input"><span class="help-tooltip search-label-icon" aria-label="Type into this box to filter down the list of selected permissions."></span></label>'
+        }
+
+        for filter_id in filter_ids:
+            filter_row = self.selenium.find_element(By.ID, f"id_permissions_{filter_id}")
+            filter_label = filter_row.find_element(By.TAG_NAME, "label")
+            self.assertEqual(filter_label.get_attribute("outerHTML"), filter_ids[filter_id])
+
+    def test_FilteredSelectMultiple_button_links(self):
+        from selenium.webdriver.common.by import By
+
+        self.admin_login(
+            username="super", password="secret", login_url=reverse("admin:index")
+        )
+        self.selenium.get(self.live_server_url + reverse("admin:auth_group_add"))
+
+        link_ids = (
+            "id_permissions_add_all_link",
+            "id_permissions_remove_all_link",
+            "id_permissions_add_link",
+            "id_permissions_remove_link"
+        )
+        
+        for link_id in link_ids:
+            link = self.selenium.find_element(By.ID, link_id)
+            self.assertEqual(link.tag_name, "button")
+
```

Comment:
Just realised there was a js_tests file for testing SelectFilter2, I'll write the tests in that before removing these